fog/sample-paykit/src/client_builder.rs

```diff
@@ -74,52 +74,52 @@ impl ClientBuilder {
     }
 
     /// Sets the grpc retry configuration
-    pub fn grpc_retry_config(self, config: GrpcRetryConfig) -> Self {
-        let mut retval = self;
-        retval.grpc_retry_config = config;
-        retval
+    #[must_use]
+    pub fn grpc_retry_config(mut self, config: GrpcRetryConfig) -> Self {
+        self.grpc_retry_config = config;
+        self
     }
 
     /// Sets the ring size to be used when generating transactions.
-    pub fn ring_size(self, ring_size: usize) -> Self {
-        let mut retval = self;
-        retval.ring_size = ring_size;
-        retval
+    #[must_use]
+    pub fn ring_size(mut self, ring_size: usize) -> Self {
+        self.ring_size = ring_size;
+        self
     }
 
     /// Sets the address book for the client, used with memos
-    pub fn address_book(self, address_book: Vec<PublicAddress>) -> Self {
-        let mut retval = self;
-        retval.address_book = address_book;
-        retval
+    #[must_use]
```

Comment:
That is cool, I knew types could be must use, I didn't know functions could also

Reply:
I also learned this, thanks to Clippy

Reply:
I think it might be worth annotating some of these with must_uses' with comments so when new engineers stumble upon it, it's more clear WHY.